pvsystemprofiler/scripts/modules/script_functions.py

```diff
@@ -143,122 +331,72 @@ def extract_sys_parameters(file_name, system, location):
-            i = file_name.rfind('/')
-            file_name = file_name[i + 1:]
-            output_list.append(file_name)
-            size_list.append(file_size)
-    if file_size_list:
-        return output_list, size_list
-    else:
-        return output_list
-
-def resume_run(output_file):
-    if os.path.isfile(output_file):
-        df = pd.read_csv(output_file, index_col=0)
-        df['site'] = df['site'].apply(str)
-        df['system'] = df['system'].apply(str)
-    else:
-       df = None
-    return df
-
-def create_system_dict(df):
-    site_list = df['site'].unique().tolist()
-    ss_dict = {}
-    for site in site_list:
-        systems_in_site = df[df['site'] == site]['system'].values.tolist()
-        ss_dict[site] = systems_in_site
-    return site_list, ss_dict
-
-def load_data(data_source, site_id):
-    for file_id in file_list:
-        file_name = file_id.split('/')[1]
-        loc = location + file_name
-        df = pd.read_csv(str(loc), index_col=0)
-
-    if data_source == 'constellation':
-        df = load_constellation_data(file_id=site_id)
-    if data_source == 'source_2':
-        df = load_cassandra_data(file_id=site_id)
-    return df
+def string_to_boolean(value):
+    """
+    Intended to be used when getting values from a terminal using `sys.argv`. Transforms `True` and `False` strings into
+    boolean attributes
+    :param value: string, `True` or `False`.
+    :return: boolean equivalent of string
+    """
+    if value in ['True', 'true']:
+        return True
+    elif value in ['False', 'false']:
+        return False
 
 
-def run_failsafe_pipeline(dh_in, df_in, sys_tag, fts, tzc):
+def run_failsafe_pipeline(df_in, manual_time_shift, sys_tag, fts, tzc):
+    """
+    Runs the solarDataTools dataHandler pipeline in failsafe mode.
+    :param manual_time_shift: Boolean. True if manual time shift inspection is performed.
+    :param df_in: Dataframe containing site input power signal.
+    :param sys_tag: Dataframe column label identifying an input signal, i.e. ac_power_01 ar dc_current_02.
+    :param fts: Boolean. Fix time shift parameter in `run_pipeline`
+    :param tzc: Boolean. Time zone correction parameter in `run_pipeline`
+    :return: Boolean. True if passes pipeline, otherwise False.
+    """
+    dh = DataHandler(df_in)
+    if manual_time_shift == 1:
+        dh.fix_dst()
     try:
         try:
-            dh_in.run_pipeline(power_col=sys_tag, fix_shifts=fts, correct_tz=tzc, verbose=False)
+            dh.run_pipeline(power_col=sys_tag, fix_shifts=fts, correct_tz=tzc, verbose=False)
         except ValueError:
             max_val = np.nanquantile(df_in[sys_tag], 0.95)
-            dh_in.run_pipeline(power_col=sys_tag, fix_shifts=False, correct_tz=tzc, verbose=False, max_val=max_val * 3)
+            dh.run_pipeline(power_col=sys_tag, fix_shifts=False, correct_tz=tzc, verbose=False, max_val=max_val * 3)
     except:
         return False
     return True
```

Comment:
need to return the data handler object here so that it is available in the outer scope